app/scripts/views/confirm_account_unlock.js

```diff
@@ -0,0 +1,166 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+define([
+  'cocktail',
+  'views/form',
+  'views/base',
+  'stache!templates/confirm_account_unlock',
+  'lib/promise',
+  'lib/auth-errors',
+  'lib/constants',
+  'views/mixins/resend-mixin',
+  'views/mixins/service-mixin',
+  'views/mixins/back-mixin'
+],
+function (Cocktail, FormView, BaseView, Template, p, AuthErrors, Constants,
+    ResendMixin, ServiceMixin, BackMixin) {
+  var t = BaseView.t;
+
+  var View = FormView.extend({
+    template: Template,
+    className: 'confirm_account_unlock',
+
+    // used by unit tests
+    VERIFICATION_POLL_IN_MS: Constants.VERIFICATION_POLL_IN_MS,
+
+    initialize: function () {
+      var data = this.ephemeralData();
+      var accountData;
+
+      if (data) {
+        if (data.lockoutSource) {
+          this._lockoutSource = data.lockoutSource;
+        }
+
+        if (data.account) {
+          accountData = data.account;
+        }
+      }
+
+      this._account = this.user.initAccount(accountData);
+    },
+
+    getAccount: function () {
+      return this._account;
+    },
+
+    context: function () {
+      return {
+        email: this.getAccount().get('email')
+      };
+    },
+
+    events: {
+      // validateAndSubmit is used to prevent multiple concurrent submissions.
+      'click #resend': BaseView.preventDefaultThen('validateAndSubmit')
+
+    },
+
+    beforeRender: function () {
+      // browsing directly to the page should not be allowed.
+      var self = this;
+      return p().then(function () {
+        if (self.getAccount().isEmpty()) {
+          self.navigate('signup');
+          return false;
+        }
+      });
+    },
+
+    afterVisible: function () {
+      var graphic = this.$el.find('.graphic');
+      graphic.addClass('pulse');
+
+      var self = this;
+      return self.broker.persist()
+        .then(function () {
+          return self._waitForConfirmation();
+        })
+        .then(function (updatedSessionData) {
+          self.getAccount().set(updatedSessionData);
+          self.logScreenEvent('verification.success');
+
+          // the continuation path depends on the action that triggerred
+          // the account lockout notice. The only time the broker should
+          // be notified is if the user was trying to sign in.
+          if (/signin/.test(self._lockoutSource)) {
```

Comment:
A regexp match seems inelegant here, but I won't be surprised if it's the least gross option